scalafmt-core/shared/src/main/scala/org/scalafmt/rewrite/SortModifiers.scala

```diff
@@ -0,0 +1,92 @@
+package org.scalafmt.rewrite
+
+import org.scalafmt.config.SortSettings._
+
+import scala.meta.Tree
+import scala.meta._
+
+object SortModifiers extends Rewrite {
+
+  override def rewrite(code: Tree, ctx: RewriteCtx): Seq[Patch] = {
+    val order = ctx.style.rewrite.sortModifiers.order
+
+    val sortMods: Seq[Mod] => Seq[Mod] = { mods =>
+      mods.sortWith(orderModsBy(order))
+    }
+
+    /*
+     * in the case of Class, Object, and of class constructor parameters
+     * some Mods are immovable, e.g. "case class X".
+     *
+     * The case of parameters is a bit more curious, because there the
+     * "val" or "var" in, say:
+     * {{{
+     *   class Test(private final val x: Int)
+     * }}}
+     * are considered Mods, instead of being similar to `Defn.Val`, or `Defn.Var`.
+     */
+    val patchesOfPatches = code.collect {
+      case d: Defn.Def => patchMods(sortMods, d.mods)
+      case v: Defn.Val => patchMods(sortMods, v.mods)
+      case v: Defn.Var => patchMods(sortMods, v.mods)
+      case t: Defn.Type => patchMods(sortMods, t.mods)
+      case c: Defn.Class => patchMods(sortMods, c.mods.filterNot(isCase))
+      case o: Defn.Object => patchMods(sortMods, o.mods.filterNot(isCase))
+      case t: Defn.Trait => patchMods(sortMods, t.mods)
+      case p: Term.Param => patchMods(sortMods, p.mods.filterNot(isValOrVar))
+    }
+    patchesOfPatches.flatten
+  }
+
+  private val isValOrVar: Mod => Boolean = m =>
+    m.is[Mod.ValParam] || m.is[Mod.VarParam]
+
+  private val isCase: Mod => Boolean = m => m.is[Mod.Case]
+
+  private def patchMods(
+      sortMods: Seq[Mod] => Seq[Mod],
```

Comment:
Why pass in the function as a parameter  instead of `sortedMods: Seq[Mod]`? Alternatively, it will always be `mods.sortWith(orderModsBy(order))` so that can be inlined at line 51 below